tokio-console/src/view/styles.rs

```diff
@@ -126,39 +146,97 @@ impl Styles {
         }
     }
 
-    pub fn time_units<'a>(&self, text: impl Into<Cow<'a, str>>) -> Span<'a> {
-        let mut text = text.into();
-        if !self.toggles.color_durations() {
-            return Span::raw(text);
-        }
+    fn duration_text(&self, dur: Duration, width: usize, prec: usize) -> FormattedDuration {
+        let secs = dur.as_secs();
+
+        if secs >= 60 * 60 * 24 * 100 {
+            let days = secs / (60 * 60 * 24);
+            FormattedDuration::Days(format!("{days:>width$}d", days = days, width = width))
+        } else if secs >= 60 * 60 * 24 {
+            let hours = secs / (60 * 60);
+            FormattedDuration::DaysHours(format!(
+                "{:>width$}",
+                format!(
+                    "{days}d{hours:02.0}h",
+                    days = hours / 24,
+                    hours = hours % 24,
+                ),
+                width = width
+            ))
+        } else if secs >= 60 * 60 {
+            let mins = secs / 60;
+            FormattedDuration::HoursMinutes(format!(
+                "{:>width$}",
+                format!(
+                    "{hours}h{minutes:02.0}m",
+                    hours = mins / 60,
+                    minutes = mins % 60,
+                ),
+                width = width
+            ))
+        } else if secs >= 60 {
+            FormattedDuration::MinutesSeconds(format!(
+                "{:>width$}",
+                format!(
+                    "{minutes}m{seconds:02.0}s",
+                    minutes = secs / 60,
+                    seconds = secs % 60,
+                ),
+                width = width,
+            ))
+        } else {
+            let mut text = format!("{:>width$.prec$?}", dur, width = width, prec = prec);
 
-        if !self.utf8 {
-            if let Some(mu_offset) = text.find("µs") {
-                text.to_mut().replace_range(mu_offset.., "us");
+            if !self.utf8 {
+                if let Some(mu_offset) = text.find("µs") {
+                    text.replace_range(mu_offset.., "us");
+                }
             }
+
+            FormattedDuration::Debug(text)
+        }
+    }
+
+    pub fn time_units<'a>(&self, dur: Duration, prec: usize, width: Option<usize>) -> Span<'a> {
```

Comment:
Style question here. An `Option<usize>` seemed like a good idea at the time, but `None` and `Some(0)` are equivalent. Do you have any opinion on the merits of just passing a bare `usize` instead?

Reply:
You could also use an `Option<NonZeroUsize>` i suppose (which has the same representation as a single `usize`, with the actual zero value representing `None`)...but this is an internal API, so misuse-resistent API design is less important. I definitely prefer `None` to indicate "no width limit" than 0, though --- when reading the code you might expect "0 width" to mean "no characters", i guess?

Reply:
Okay. That sounds like a better option (using NonZero). Internal or no, as you said, None to mean no width restriction makes more sense than 0. I'll document the function too, so that it's clear. Thanks!

Reply:
sounds good to me!

Reply:
ugh, that's annoying. i would prefer not to put unsafe code in the const. at the callsite, we could call `time_units(..., NonZeroUsize::new(DUR_LEN))` but that's more verbose. i think it would be fine to allow `Some(0)` and `None` to mean the same thing, or just get rid of the option...

Reply:
Changed it back to `Option<usize>`.